snap/snapcraft.yaml

```diff
@@ -0,0 +1,82 @@
+name: ghostty
+base: core24
+summary: A terminal emulator
+description: |
+  Ghostty is a fast, feature-rich, and cross-platform terminal emulator that
+  uses platform-native UI and GPU acceleration.
+grade: stable
+confinement: classic
+contact: https://github.com/ghostty-org/ghostty/discussions
+issues: https://github.com/ghostty-org/ghostty/issues
+website: https://ghostty.org
+license: MIT
+icon: images/icons/icon_512.png
+adopt-info: ghostty
+
+platforms:
+  amd64:
+  arm64:
+
+apps:
+  ghostty:
+    command: bin/ghostty
+    command-chain: [ bin/launcher ]
+    completer: share/bash-completion/completions/ghostty.bash
+    desktop: share/applications/com.mitchellh.ghostty.desktop 
+    environment:
+      PATH: /snap/ghostty/current/bin:/snap/ghostty/current/usr/bin:$PATH
+      LC_ALL: C.UTF-8
+      GHOSTTY_RESOURCES_DIR: /snap/ghostty/current/share/ghostty
+
+parts:
+  launcher:
+    plugin: dump
+    source: snap/local
+    source-type: local
+    organize:
+      launcher: bin/
+
+  zig:
+    plugin: nil
+    build-packages:
+      - curl
+    override-pull: |
+      set -ex
+      case "$CRAFT_TARGET_ARCH" in
+        amd64)  arch=x86_64    ;;
+        arm64)  arch=aarch64  ;;
+        *)      arch=""     ;;
+      esac
+
+      rm -rf $CRAFT_PART_SRC/*
+
+      if [[ -n $arch ]]; then
+        curl -LO --retry-connrefused --retry 10 https://ziglang.org/download/0.13.0/zig-linux-$arch-0.13.0.tar.xz
+      else
+        echo "Unsupported arch"
+        exit 1
+      fi
+
+      tar xf zig-lin*xz
+      rm -f *xz
+      mv zig-linux*/* .
+    prime:
+      - -*
+
+  ghostty:
```

Comment:
I wonder why you did not choose to use the pre-built packages available at https://github.com/ghostty-org/ghostty/blob/main/PACKAGING.md?

Reply:
That's more desirable for third party packagers, but if the ghostty project is publishing the snap it's nice to keep the packaging in the source and stable releases can be easily maintained by building from stable branches or specific tags.